src/components/ConnectButton.tsx

```diff
@@ -0,0 +1,128 @@
+import { walletsAndConnectors } from "@/constants";
+import { ConnectButton as RainbowkitConnectButton } from "@rainbow-me/rainbowkit";
+import NextImage from "next/image";
+import Chevron from "public/assets/chevron.svg";
+import { useEffect, useState } from "react";
+import styled, { CSSProperties } from "styled-components";
+import { useAccount } from "wagmi";
+
+export function ConnectButton() {
+  const { connector } = useAccount();
+  const [walletIcon, setWalletIcon] = useState("");
+
+  const wallets = walletsAndConnectors.wallets.flatMap(
+    ({ wallets }) => wallets
+  );
+
+  const iconsAndIds = wallets.map(({ id, iconBackground, iconUrl }) => ({
```

Comment:
for some reason the icons for specific wallets are not readily available from the `accounts`. I think it's because most people prefer to show ens avatars instead of the wallet logo like we do.

this gets the wallet icons from where rainbowkit stores them, and finds that appropriate icon for the connected wallet by matching on the id field, which is present in both.